### ISSUE TITLE
{:network_failure, "smtp.my_server.com", {:error, :timeout}}

### ISSUE THREAD

**Issue by Douvi** [fewlinesco/bamboo_smtp#52]
Hello,

I have a little issue with your lib. After setup Bamboo into my project see below the conf :

```
config :test_phoenix, TestPhoenix.Mailer,
        adapter: Bamboo.SMTPAdapter,
        server: "smtp.my_server.com",
        port: 25,
        username: "user",
        password: "paSSwork",
        tls: :never, # can be `:always` or `:if_available`
        ssl: false # can be `true`
```

I got this error:

```
[error] Task #PID<0.1100.0> started from #PID<0.1098.0> terminating
** (Bamboo.SMTPAdapter.SMTPError) There was a problem sending the email through SMTP.

The error is :retries_exceeded

More detail below:

{:network_failure, "smtp.my_server.co", {:error, :timeout}}

    lib/bamboo/adapters/smtp_adapter.ex:74: Bamboo.SMTPAdapter.handle_response/1
    (elixir) lib/task/supervised.ex:94: Task.Supervised.do_apply/2
    (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
Function: #Function<0.115485522/0 in Bamboo.TaskSupervisorStrategy.deliver_later/3>
    Args: []
```

So I check online if I can send an email with `http://smtper.nanogenesis.fr/`
with the same param. And I got the email!!!

So I do not know what I did wrong!!!

PS: The server email is `postfix version 2.6.6`

**Comment by tgautier** [comment]
You're getting a `timeout` error which means that `gen_smtp` can't connect to the server with your configuration.

I'm not sure we can be of any help.

Have you triple-checked that you have the correct configuration?
Have you tried switching the `:tls` and `:ssl` options?
Have you tried increasing `retries:`?

**Comment by rohandaxini** [comment]
@Douvi Is this issue resolved? We are facing same issues and we tried `tls` as well as `ssl` option and still no luck @tgautier 
We even upgraded bamboo_smtp to 1.4.0 and still no luck. We are facing exact same error as `retries_exceeded`. We checked our configuration several times and also modified it many times to try out variations. Earlier this was all working fine, until we enabled SSL on our server. We have just configured SSL via nginx for our Elixir Phoenix app and since then the emails have stopped working.
